### PR TITLE
Major refactor and unit tests

### PR DESCRIPTION
**Pull request by idjaw**
- The overall structure of the code was re-worked to make the code easier to read, maintain and add new features
- Modules are now easier to manage as they exist in their own module files
- A base api was introduced to provide for the base functionality of the API
- unit tests were added for all modules to ensure expected functionality
- py27 support was removed as it is unnecessary

### CODE REVIEW

File: fake_ubersmith/api/base.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+from flask import request
+
+
+class FakeUbersmithError:
+    def __init__(self, code=None, message=None):
+        self.code = code
+        self.message = message
+
+
+class BaseAPI:
```
Comment (mbelang) [inline fake_ubersmith/api/base.py]: Make this abc so that you force the inheritance in the sub api class.

File: fake_ubersmith/api/base.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    def __init__(self):
+        self.app = None
+        self.post_methods = {}
+
+    def hook_to(self, server):
```
Comment (mbelang) [inline fake_ubersmith/api/base.py]: abstractmethod it should be

File: fake_ubersmith/api/client.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from fake_ubersmith.api.base import FakeUbersmithError
+from fake_ubersmith.api.helpers import record
+from fake_ubersmith.api.response import response
+
+
+class Client:
```
Comment (mbelang) [inline fake_ubersmith/api/client.py]: Make it derive from BaseApi

File: fake_ubersmith/api/order.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from fake_ubersmith.api.base import FakeUbersmithError
+from fake_ubersmith.api.helpers import record
+from fake_ubersmith.api.response import response
+
+
+class Order:
```
Comment (mbelang) [inline fake_ubersmith/api/order.py]: Derive from BaseApi

File: fake_ubersmith/api/uber.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from fake_ubersmith.api.base import FakeUbersmithError
+from fake_ubersmith.api.helpers import record
+from fake_ubersmith.api.response import response
+
+
+class Uber:
```
Comment (mbelang) [inline fake_ubersmith/api/uber.py]: Make it derive from BaseApi

File: tox.ini [hunk 6]
```diff
@@ -1,5 +1,5 @@
 [tox]
-envlist = py27, py34, py35, flake8
+envlist = py34, py35, flake8
```
Comment (mbelang) [inline tox.ini]: You forgot to remove the job from travis.yml

Comment (idjaw) [inline tox.ini]: Addressed. Thank you.

**Pull request merged**